### PR TITLE
add imaginAIry

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
```
The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested setuptools_rust
```

**Comment by jan-janssen** [comment]
```
The reported errors are:
- Encountered problems while solving:
-   - nothing provides __cuda needed by pytorch-1.11.0-cuda102py310hdf4a2db_202
```

**Comment by jan-janssen** [comment]
Somehow the dependencies are still broken:
```
- Encountered problems while solving:
-   - package safetensors-0.2.8-py38h9fda977_0 requires python_abi 3.8.* *_cp38, but none of the providers can be installed
- 
```

**Comment by jan-janssen** [comment]
```
- Encountered problems while solving:
-   - package facexlib-0.2.5-py38h6f1a3b6_0 requires python_abi 3.8.* *_cp38, but none of the providers can be installed
```

**Comment by jan-janssen** [comment]
```
- Encountered problems while solving:
-   - nothing provides __cuda needed by fairscale-0.4.6-cuda102py310h4c99be8_5
```

**Comment by jan-janssen** [comment]
I found one issue `imaginAIry` has  `- protobuf !=3.20.2,!=3.19.5` this is in conflict with `open-clip-torch` which has `- protobuf ==3.20.*` on conda-forge there is only one `protobuf=3.20.2` release for `3.20` so both conditions can not be fulfilled at the same time.

**Comment by jan-janssen** [comment]
```
pytorch-lightning 1.8.1 has requirement lightning-utilities==0.3.*, but you have lightning-utilities 0.6.0.post0.
```

**Comment by jan-janssen** [comment]
```
ImportError: cannot import name 'raw_input' from 'click._compat'
```

File: recipes/imaginairy/meta.yaml [hunk 1]
```diff
@@ -56,7 +56,8 @@ test:
   requires:
     - pip
   commands:
-    - pip check
+    # - pip check  # disable pip check as it is not supported by pytorch-lightning
```
[the remark this answers is not in the document]
Comment (jan-janssen) [reply recipes/imaginairy/meta.yaml:59]: The issues is fixed in https://github.com/conda-forge/pytorch-lightning-feedstock/pull/115

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
No windows a couple of dependencies are missing: 
```
- Encountered problems while solving:
-   - nothing provides requested fairscale >=0.4.4
-   - nothing provides requested torchvision >=0.13.1
-   - nothing provides requested pytorch >=1.2.0
-   - nothing provides pytorch needed by facexlib-0.2.5-pyha21a80b_0
-   - nothing provides pytorch >=1.4 needed by timm-0.4.12-pyhd8ed1ab_0
-   - nothing provides pytorch >=1.9.0 needed by open-clip-torch-2.10.1-pyhd8ed1ab_0
-   - nothing provides pytorch needed by transformers-4.19.2-pyhd8ed1ab_0
-   - nothing provides pytorch >=1.3.1 needed by torchmetrics-0.10.0-pyhd8ed1ab_0
-   - nothing provides pytorch >=1.6 needed by pytorch-lightning-1.4.2-pyhd8ed1ab_0
-   - nothing provides pytorch >=1.10 needed by kornia-0.6.1-pyhd8ed1ab_0
-   - nothing provides pytorch >=1.4 needed by diffusers-0.1.2-pyhd8ed1ab_0
-   - nothing provides pytorch needed by torchdiffeq-0.2.2-pyhd8ed1ab_0
```
Still I would like this package to remain `noarch`

**Review by dopplershift — COMMENTED** [review]
Ignoring noarch Windows build failure due to missing dependency.

**Comment by brycedrennan** [comment]
@jan-janssen thanks for working through all that to get this included in conda

**Comment by jan-janssen** [comment]
@brycedrennan Sure, I am happy to - if you are interested, I can also add you as a maintainer to the conda recipe, it is always great to have the core developers on board.

**Comment by brycedrennan** [comment]
That would be great. I'm not well-versed in the conda ecosystem but would be nice to have the permission to fix any problems that may come up.

**Pull request merged**